source/lib/vagrant-openstack-provider/client/keystone.rb

```diff
@@ -96,7 +96,7 @@ def get_body_3(config)
                 user: {
                   name: config.username,
                   domain: {
-                    name: config.domain_name
```

Comment:
For backward compatibility, implementation of `get_body_3` should check if `config.domain_name` is set. If yes, it will be used for both user and project. If not, we use `config.user_domain_name` and `config.project_domain_name`.

Reply:
domain_name is now overriding user_domain_name and project_domain_name in config.rb line 376.

Validation checks messages updated accordingly.